packages/liveblocks-react/src/room.tsx

```diff
@@ -2500,45 +2544,45 @@ function useThreadsSuspense<M extends BaseMetadata>(
     [room, options]
   );
 
-  const { store, getThreadsAndInboxNotifications } =
-    getExtrasForClient<M>(client);
-
-  const query = store.getThreads().queries2[queryKey];
+  const { store, loadRoomThreads } = getExtrasForClient<M>(client);
 
-  if (query === undefined || query.isLoading) {
-    throw getThreadsAndInboxNotifications(room, queryKey, options);
-  }
+  const loadThreadsPromise = loadRoomThreads(room, queryKey, options);
 
-  if (query.error) {
-    throw query.error;
-  }
+  use(loadThreadsPromise);
 
   const selector = React.useCallback(
     (state: ReturnType<typeof store.getThreads>): ThreadsAsyncSuccess<M> => {
+      if (
+        !("status" in loadThreadsPromise) ||
+        loadThreadsPromise.status !== "fulfilled"
+      ) {
+        console.warn("Internal error: Unexpected promise status");
+      }
+
       return {
+        isLoading: false,
         threads: selectThreads(state, {
           roomId: room.id,
           query: options.query,
           orderBy: "age",
         }),
-        isLoading: false,
       };
```

Comment:
Here, we expect this selector to run only if the promise resolved correctly.

Reply:
Yep! After `use(loadThreadsPromise)`, I'd indeed expect the promise to be in `fulfilled` state. That's exactly what `use()` guarantees, so this makes sense to me. AFAIAC, you can remove this if-check and just leave a comment that at this point in time, the promise will always be resolved.